src/output/plugins/OSXOutputPlugin.cxx

```diff
@@ -685,8 +685,24 @@ osx_render(void *vdata,
 		od->cancel = false;
 	}
 
-	// Wait until data is available.
-	while (!od->ring_buffer->read_available()); 
+	// If data is not available, fill the buffer with silence,
+	if (!od->ring_buffer->read_available()) {
+#ifdef ENABLE_DSD
+		if (od->dop_enabled) {
+			const auto silence = od->pcm_export->GetSilence();
+			size_t count = in_number_frames * od->asbd.mBytesPerFrame / silence.size * silence.size;
+			for (size_t i = 0; i < count; ++i) {
+				memcpy((uint8_t *)buffer_list->mBuffers[0].mData + i * silence.size, silence.data, silence.size);
```

Comment:
This assumes that previously, everything was already written at "block" boundary; but what if the previous `osx_render()` call wrote only the first frame of a DoP "block"?

Reply:
See https://github.com/MusicPlayerDaemon/MPD/pull/582/files/3744a59da56b07fb5b4cccbf17d0787c3c1ef5e0..b6bcb97a527de660e6cb5cc68a70b077ded00182#diff-ce7ecec9ea9ca3df90d9c290cb3ef9d4R713

Reply:
the osx_render function always write complete dop blocks, so we can safely write silence here.

Reply:
Oh, good! ALSA can't do that, unfortunately, because a period can theoretically have an odd number of frames.

Reply:
I'll mark it as resolved.